.dotnet.azure/sdk/openai/Azure.AI.OpenAI/tests/VectorStoreTests.cs

```diff
@@ -181,26 +189,26 @@ public async Task CanAssociateFiles()
     public async Task CanUseBatchIngestion()
     {
         VectorStoreClient client = GetTestClient();
-        VectorStore vectorStore = await client.CreateVectorStoreAsync();
+        CreateVectorStoreOperation createVectorStoreOperation = await client.CreateVectorStoreAsync(false);
+        VectorStore vectorStore = createVectorStoreOperation.GetVectorStore();
         Validate(vectorStore);
 
         IReadOnlyList<OpenAIFileInfo> testFiles = await GetNewTestFilesAsync(client.GetConfigOrThrow(), 3);
 
-        VectorStoreBatchFileJob batchJob = await client.CreateBatchFileJobAsync(vectorStore, testFiles);
+        CreateBatchFileJobOperation createBatchFileJobOperation = await client.CreateBatchFileJobAsync(vectorStore, testFiles, false);
+        VectorStoreBatchFileJob batchJob = createBatchFileJobOperation.GetFileBatch();
         Assert.Multiple(() =>
         {
             Assert.That(batchJob.BatchId, Is.Not.Null);
             Assert.That(batchJob.VectorStoreId, Is.EqualTo(vectorStore.Id));
             Assert.That(batchJob.Status, Is.EqualTo(VectorStoreBatchFileJobStatus.InProgress));
         });
 
-        batchJob = await WaitUntilReturnLast(
-            batchJob,
-            () => client.GetBatchFileJobAsync(batchJob),
-            b => b.Status != VectorStoreBatchFileJobStatus.InProgress);
+        createBatchFileJobOperation.WaitForCompletion();
+        batchJob = createBatchFileJobOperation.GetFileBatch();
         Assert.That(batchJob.Status, Is.EqualTo(VectorStoreBatchFileJobStatus.Completed));
 
-        AsyncPageCollection<VectorStoreFileAssociation> response = client.GetFileAssociationsAsync(batchJob);
+        AsyncPageCollection<VectorStoreFileAssociation> response = client.GetFileAssociationsAsync(vectorStore);
```

Comment:
Could we get this collection from the job itself instead of the client?

Reply:
It seems we could. I don't know the vector store scenario but the comments on the methods seem close enough to me. I'll update the test to use the operation